### PR TITLE
Loadouts: option to put magazine in weapon directly

### PR DESCRIPTION
**Pull request by Boberro**
Addds new attributes in loadouts: primaryMagazine, primaryGrenade, secondaryMagazine and sidearmMagazine, as suggested in #181 
A workaround to #263 

Additionally adds loadout index defines to assign gear pbo script_component - for sane changing / reordering of cached loadout attributes in the future.

Example use:
`secondaryMagazine[] = {"rhs_rpg7_PG7VL_mag", "rhs_rpg7_PG7VR_mag"}; // adds random rocket`
`secondaryMagazine[] = {"rhs_rpg7_PG7VL_mag", "default"}; // either adds additional rocket, or takes default from magazines[]`

It appears that use of "" for spawning unit with no magazine in weapon is impossible - it works as intended in Eden, but in actual play player still spawns with weapon reloaded.

### CODE REVIEW

File: addons/assigngear/functions/fn_assignGear.sqf [hunk 1]
```diff
@@ -76,81 +99,108 @@ _unit setUnitLoadout (configFile >> 'EmptyLoadout');
...
+                private _magazine = selectRandom _x;
+                if (toLower _magazine != 'default') then {
+                    private _backup = secondaryWeaponMagazine _unit;
+                    _unit addSecondaryWeaponItem _magazine;
+                    [_backup] call _fnc_addMagazines;
+                };
+            };
+            case IDX_SIDEARM_MAGAZINE: {
+                private _magazine = selectRandom _x;
+                if (toLower _magazine != 'default') then {
```
Comment (Freddo3000) [inline addons/assigngear/functions/fn_assignGear.sqf]: `!=` and `==` are case-insensitive

For case sensitive comparison, you use `isEqualTo`

Comment (Boberro) [reply addons/assigngear/functions/fn_assignGear.sqf]: well then, reverting this one change

**Comment by Boberro** [comment]
Added fix to #298 as it'd collide with Freddo's fix.

**Comment by Boberro** [comment]
eg. I use it in adversarials, to create single loadout for both sides.

**Comment by Boberro** [comment]
And here we go again

**Pull request closed without merging**